lib/std/x/os/socket_posix.zig

```diff
@@ -0,0 +1,235 @@
+        return os.listen(self.fd, max_backlog_size);
+    }
+
+    /// Have the socket attempt to the connect to an address.
+    pub fn connect(self: Socket, address: Socket.Address) !void {
+        return os.connect(self.fd, @ptrCast(*const os.sockaddr, &address.toNative()), address.getNativeSize());
+    }
+
+    /// Accept a pending incoming connection queued to the kernel backlog
+    /// of the socket.
+    pub fn accept(self: Socket, flags: u32) !Socket.Connection {
+        var address: os.sockaddr_storage = undefined;
+        var address_len: u32 = @sizeOf(os.sockaddr_storage);
+
+        const socket = Socket{ .fd = try os.accept(self.fd, @ptrCast(*os.sockaddr, &address), &address_len, flags) };
+        const socket_address = Socket.Address.fromNative(@alignCast(4, @ptrCast(*os.sockaddr, &address)));
+
+        return Socket.Connection.from(socket, socket_address);
+    }
+
+    /// Read data from the socket into the buffer provided with a set of flags
+    /// specified. It returns the number of bytes read into the buffer provided.
+    pub fn read(self: Socket, buf: []u8, flags: u32) !usize {
+        return os.recv(self.fd, buf, flags);
+    }
+
+    /// Write a buffer of data provided to the socket with a set of flags specified.
+    /// It returns the number of bytes that are written to the socket.
+    pub fn write(self: Socket, buf: []const u8, flags: u32) !usize {
+        return os.send(self.fd, buf, flags);
+    }
+
+    /// Writes multiple I/O vectors with a prepended message header to the socket
+    /// with a set of flags specified. It returns the number of bytes that are
+    /// written to the socket.
+    pub fn writeVectorized(self: Socket, msg: os.msghdr_const, flags: u32) !usize {
+        return os.sendmsg(self.fd, msg, flags);
+    }
+
+    /// Read multiple I/O vectors with a prepended message header from the socket
+    /// with a set of flags specified. It returns the number of bytes that were
+    /// read into the buffer provided.
+    pub fn readVectorized(self: Socket, msg: *os.msghdr, flags: u32) !usize {
+        return error.NotImplemented;
+    }
+
+    /// Query the address that the socket is locally bounded to.
+    pub fn getLocalAddress(self: Socket) !Socket.Address {
+        var address: os.sockaddr_storage = undefined;
+        var address_len: u32 = @sizeOf(os.sockaddr_storage);
+        try os.getsockname(self.fd, @ptrCast(*os.sockaddr, &address), &address_len);
+        return Socket.Address.fromNative(@alignCast(4, @ptrCast(*os.sockaddr, &address)));
+    }
+
+    /// Query the address that the socket is connected to.
+    pub fn getRemoteAddress(self: Socket) !Socket.Address {
+        var address: os.sockaddr_storage = undefined;
+        var address_len: u32 = @sizeOf(os.sockaddr_storage);
+        try os.getpeername(self.fd, @ptrCast(*os.sockaddr, &address), &address_len);
+        return Socket.Address.fromNative(@alignCast(4, @ptrCast(*os.sockaddr, &address)));
```

Comment:
Can you not do the align in the ptrCast?
`@ptrCast(*align(4) os.sockaddr, ...)`

Reply:
I'll double check tomorrow,

Reply:
Found out that I can directly @ptrCast to *os.sockaddr now that os.sockaddr_storage is used. Fix will appear in the next commit.

Reply:
Fix landed.